src/web-animations-next-animation.js

```diff
@@ -25,13 +27,48 @@
     this._animation = null;
     this._childAnimations = [];
     this._callback = null;
+    this._oldPlayState = 'idle';
     this._rebuildUnderlyingAnimation();
     // Animations are constructed in the idle state.
     this._animation.cancel();
+    this._updatePromises();
   };
 
   scope.Animation.prototype = {
+    _updatePromises: function() {
+      var oldPlayState = this._oldPlayState;
+      var newPlayState = this.playState;
+      if (this._readyPromise && newPlayState !== oldPlayState) {
+        if (newPlayState == 'idle') {
+          if (this._readyPromiseState == 'pending') {
```

Comment:
> You might be able to simplify the _{ready,reject}PromiseState checks this way.

We don't even need to store promise state if we don't check it here. Although I don't know how I would test this if I don't store promise state.